sky/compositor/picture_layer.cc

```diff
@@ -0,0 +1,114 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "sky/compositor/picture_layer.h"
+#include "base/logging.h"
+#include "third_party/skia/include/gpu/GrContext.h"
+#include "third_party/skia/include/core/SkSurface.h"
+
+namespace sky {
+namespace compositor {
+
+PictureLayer::PictureLayer() : last_picture_id_(UINT32_MAX) {
+}
+
+PictureLayer::~PictureLayer() {
+}
+
+SkMatrix PictureLayer::model_view_matrix(const SkMatrix& model_matrix) const {
+  SkMatrix modelView = model_matrix;
+  modelView.postTranslate(offset_.x(), offset_.y());
+  return modelView;
+}
+
+static void ImageReleaseProc(SkImage::ReleaseContext texture) {
+  DCHECK(texture);
+  reinterpret_cast<GrTexture*>(texture)->unref();
+}
+
+static RefPtr<SkImage> ImageFromPicture(GrContext* context,
+                                        SkPicture* picture,
+                                        const SkRect& paintBounds) {
+  // Step 1: Create a texture from the context's texture provider
+
+  GrSurfaceDesc desc;
+  desc.fWidth = paintBounds.width();
+  desc.fHeight = paintBounds.height();
```

Comment:
Don't we need to scale the size of the texture we produce by the matrix?  Maybe you're planning to do that in a later patch.